apps/src/templates/sectionProgressV2/OpenAllStudentProgressButton.jsx

```diff
@@ -0,0 +1,149 @@
+
+import FontAwesome from '../FontAwesome';
+import {
+  collapseMetadataForStudents,
+  expandMetadataForStudents,
+} from '../sectionProgress/sectionProgressRedux';
+
+import style from './expand-all-rows-dropdown.module.scss';
+
+const DROPDOWN_OFFSET = 172;
+
+function OpenAllStudentProgressButton({
+  students,
+  expandMetadataForStudents,
+  collapseMetadataForStudents,
+}) {
+  const [opened, setOpened] = useState(false);
+  const [menuLocation, setMenuLocation] = useState({menuTop: 0, menuLeft: 0});
+  const elementRef = useRef(null);
+  const resizeListener = useRef(null);
+
+  const getAllStudentIds = students.map(student => student.id);
+
+  const expandMetaDataForAllStudents = () => {
+    expandMetadataForStudents(getAllStudentIds);
+    setOpened(false);
+  };
+
+  const collapseMetaDataForAllStudents = () => {
+    collapseMetadataForStudents(getAllStudentIds);
+    setOpened(false);
+  };
+
+  const getMenuLocation = () => {
+    const rect = elementRef.current.firstChild.getBoundingClientRect();
+    console.log(
+      rect.bottom,
+      window.pageYOffset,
+      rect.right,
+      window.pageXOffset
+    );
+    return {
+      menuTop: rect.bottom + window.pageYOffset,
+      menuLeft: rect.right + window.pageXOffset - DROPDOWN_OFFSET,
+    };
+  };
+
+  const updateMenuLocation = throttle(() => {
+    setMenuLocation(getMenuLocation());
+  }, 50);
+
+  useEffect(() => {
+    if (opened && !resizeListener.current) {
+      resizeListener.current = updateMenuLocation;
+      window.addEventListener('resize', resizeListener.current);
+      setMenuLocation(getMenuLocation());
+    } else if (!opened && resizeListener.current) {
+      window.removeEventListener('resize', resizeListener.current);
+      resizeListener.current = null;
+    }
+
+    return () => {
+      if (resizeListener.current) {
+        window.removeEventListener('resize', resizeListener.current);
+      }
+    };
+  }, [updateMenuLocation, opened]);
+
+  const handleDropdownClick = () => {
+    setOpened(!opened);
+  };
+
+  const closeMenu = () => {
+    if (opened) setOpened(false);
+  };
+
+  const onClose = () => {
+    closeMenu();
+  };
+
+  const targetPoint = {top: menuLocation.menuTop, left: menuLocation.menuLeft};
+
+  return (
+    <div className={style.main} ref={elementRef}>
+      <button
+        type="button"
+        className={style.expandButton}
+        onClick={handleDropdownClick}
+        aria-label={i18n.additionalOptions()}
+      >
+        <FontAwesome icon="ellipsis-vertical" />
+      </button>
+      {opened && (
+        <PopUpMenu
+          isOpen={opened}
+          targetPoint={targetPoint}
+          offset={{x: -elementRef.current.firstChild.offsetWidth, y: 5}}
+          onClose={onClose}
+          className={style.menu}
+        >
+          <PopUpMenu.Item
+            className={style.menuItem}
+            onClick={expandMetaDataForAllStudents}
+            style={{marginTop: '4px', padding: '5px 14px'}}
```

Comment:
Can we use SCSS instead of style?

Reply:
Just to clarify, do you mean add this style to the className `menuItem`?  I tried that initially but it wasn't working.  When I looked at the `PopUpMenu` component I saw that `style` was its own prop (in addition to `className`).  When I put the styles I wanted into a class for `scss` it wouldn't work, however, if I added it in `style` directly as a prop, it worked (this made sense with what I saw in `PopUpMenu` as well).  I tried adding the styles to the `scss` file and then using it as a class in `style`... 

... this is getting long, BUT, through this I learned that `style` needs an object with CSS properties, and when passing through a class, it gets a string.  I got into a way to extract the CSS from the `scss` file, but that made the code way less readable.